File: backend/domain/table/delivery.go
package table

import (
	"time"

	z "github.com/Oudwins/zog"
)

type Delivery struct {
	ID          string         `json:"id"`
	UserID      int            `json:"userId"`
	TableID     int            `json:"tableId"`
	Variants    []LineItem `json:"variants"`
	Comment     string         `json:"comment"`
	DeliveredAt time.Time      `json:"deliveredAt"`
}

var deliverySchema = z.Struct(z.Shape{
	"ID":          z.String().UUID().Required(),
	"UserID":      z.Int().GTE(1).Required(),
	"TableID":     z.Int().GTE(1).Required(),
	"Variants":    z.Slice(lineItemSchema).Min(1).Required(),
	"Comment":     z.String().Max(100),
	"DeliveredAt": z.Time().Required(),
})

File: frontend/src/service/table/Delivery.ts
import { z } from 'zod'

import { OrderVariantSchema } from './Order'

export const DeliverySchema = z.object({
  id: z.uuid(),
  userId: z.number().int().min(1),
  tableId: z.number().int().min(1),
  variants: OrderVariantSchema.array().min(1),
  comment: z.string().max(100),
  deliveredAt: z.string().refine((date) => !isNaN(Date.parse(date)), {
    message: 'Invalid date format',
  }),
})
export type Delivery = z.infer<typeof DeliverySchema>

export const DeliverVariantsSchema = z.object({
  tableId: z.number().int().min(1),
  variants: OrderVariantSchema.array().min(1),
  comment: z.string().max(100),
})
export type DeliverVariants = z.infer<typeof DeliverVariantsSchema>
